public/app/features/alerting/unified/components/rules/central-state-history/EventListSceneObject.tsx

```diff
@@ -423,10 +548,16 @@ function useRuleHistoryRecords(stateHistory?: DataFrameJSON, filter?: string) {
       if (!isLine(line)) {
         return acc;
       }
-
       // values property can be undefined for some instance states (e.g. NoData)
       const filterMatch = line.labels && labelsMatchMatchers(line.labels, filterMatchers);
-      if (filterMatch) {
+      if (!isGrafanaAlertState(line.current) || !isGrafanaAlertState(line.previous)) {
+        return acc;
+      }
+      const baseStateTo = mapStateWithReasonToBaseState(line.current);
+      const baseStateFrom = mapStateWithReasonToBaseState(line.previous);
+      const stateToMatch = filterInStateTo !== StateToFilterValues.all ? filterInStateTo === baseStateTo : true;
+      const stateFromMatch = filterInStateFrom !== StateToFilterValues.all ? filterInStateFrom === baseStateFrom : true;
```

Comment:
This is referencing `StateToFilterValues` here rather than `StateFromFilterValues`, but I feel that we should have one type for those anyway, as (for now), we always allow the same options for `from` and `to`?

Reply:
I agree, let's keep one constant for both, as for now are the same.